jgrapht-core/src/main/java/org/jgrapht/alg/cycle/ChordalGraphMinimalVertexSeparatorFinder.java

```diff
@@ -0,0 +1,167 @@
+/*
+ * (C) Copyright 2018-2018, by Timofey Chudakov and Contributors.
+ *
+ * JGraphT : a free Java graph-theory library
+ *
+ * This program and the accompanying materials are dual-licensed under
+ * either
+ *
+ * (a) the terms of the GNU Lesser General Public License version 2.1
+ * as published by the Free Software Foundation, or (at your option) any
+ * later version.
+ *
+ * or (per the licensee's choosing)
+ *
+ * (b) the terms of the Eclipse Public License v1.0 as published by
+ * the Eclipse Foundation.
+ */
+package org.jgrapht.alg.cycle;
+
+import org.jgrapht.Graph;
+import org.jgrapht.Graphs;
+
+import java.util.*;
+
+/**
+ * Allows obtaining a mapping of all <a href="https://en.wikipedia.org/wiki/Chordal_graph#Minimal_separators">
+ * minimal vertex separators</a> of a graph to their multiplicities
+ * <p>
+ * In the context of this implementation following definitions are used:
+ * <ul>
+ * <li>A set of vertices $S$ of a graph $G=(V, E)$ is called a <i>u-v separator</i>, if vertices $u$ and
+ * $v$ in the induced graph on vertices $V(G) - S$ are in different connected components.</li>
+ * <li>A set $S$ is called a <i>minimal u-v separator</i> if it is a u-v separator and no proper subset of $S$ is
+ * a u-v separator.</li>
+ * <li> A set $S$ is called a <i>minimal vertex separator</i> if it is minimal u-v separator for some
+ * vertices $u$ and $v$ of the graph $G$.</li>
+ * <li>A set of vertices $S$ is called a <i>minimal separator</i> if no proper subset of $S$ is
+ * a separator of the graph $G$.</li>
+ * </ul>
+ * <p>
+ * Let $\sigma = (v_1, v_2, \dots, v_n)$ be some perfect elimination order (peo) of the graph $G = (V, E)$.
```

Comment:
Let's add the following:

> The multiplicity of a minimal vertex separator indicates the number of different pairs of vertices separated by it.

I copy/pasted this from the paper you were referencing; I had never heard of the multiplicity of a vertex separator before.